### PR TITLE
Add ADF readme

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Missing readme in previous PRs.

### 🤓 What?

- Add the readme for ADF connector

### 🧪 Tested?

Verified we can limit permission for only some of data factories and we the metadata we get is correct.

### CODE REVIEW

File: metaphor/azure_data_factory/readme.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+This connector extracts technical metadata from Azure Data Factories using [Azure REST API](https://learn.microsoft.com/en-us/rest/api/datafactory/v2)
+
+## Setup
+
+We recommend creating a dedicated Azure AD Application for the connector to use.
+
+1. Follow [Step 1 of this doc](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal#step-1---create-an-azure-ad-app) to create an Azure AD app and a client secret.
+
+2. Give the app a reader role to access REST APIs. Go to the resource you want to connect to. For example, a subscription. On the resource page, select `Access Control (IAM)`, and add the built-in `Reader` role to the app you created in Step 1.
```
Comment (mars-lan) [inline metaphor/azure_data_factory/readme.md:11]: I'm not sure I follow this completely. So is the idea to give global READ permission to the subscription the ADF belongs to? Can we do it at a finer grain?

File: metaphor/azure_data_factory/readme.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+### Required Configurations
+
+```yaml
+tenant_id: <tenant_id>  # The Azure Directory (tenant) ID
+
+client_id: <client_id>  # The Azure Application client id
+
+client_secret: <client_secret>  # The client secret value (not secret ID)
+
+subscription_id: <subscription_id>  # Azure subscription id
```
Comment (mars-lan) [inline metaphor/azure_data_factory/readme.md:26]: What if there are multiple Azure data factories using different subscription IDs? Do we need to create one for each subscription?

File: metaphor/azure_data_factory/readme.md (hunk 1 above)
Comment (elic-eon) [reply metaphor/azure_data_factory/readme.md:11]: Yes, you're right. It's a global reader role. There are no built-in read-only role for ADF, we can provide a role permission in the document as well.

File: metaphor/azure_data_factory/readme.md (hunk 2 above)
Comment (elic-eon) [reply metaphor/azure_data_factory/readme.md:26]: Yes, we will need to create one for each subscription.

File: metaphor/azure_data_factory/readme.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ We recommend creating a dedicated Azure AD Application for the connector to use.
 
 1. Follow [Step 1 of this doc](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal#step-1---create-an-azure-ad-app) to create an Azure AD app and a client secret.
 
-2. Give the app a reader role to access REST APIs. Go to the resource you want to connect to. For example, a subscription. On the resource page, select `Access Control (IAM)`, and add the built-in `Reader` role to the app you created in Step 1.
+2. Give the app a reader role for each data factory to access REST APIs. In the Azure portal, search `Data Factories`, click the data factory you want to connect to. Select `Access Control (IAM)`, and add the built-in `Reader` role to the app you created in Step 1.
```
Comment (mars-lan) [inline metaphor/azure_data_factory/readme.md:11]:
```suggestion
2. Give the app the [Reader Role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) for each data factory to access the REST APIs. Select `Access Control (IAM)` from each data factory's settings page and add the built-in `Reader` role to the app you created in Step 1.
```

File: metaphor/azure_data_factory/readme.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+This connector extracts technical metadata from Azure Data Factories using [Azure REST API](https://learn.microsoft.com/en-us/rest/api/datafactory/v2)
+
+## Setup
+
+We recommend creating a dedicated Azure AD Application for the connector to use.
+
+1. Follow [Step 1 of this doc](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal#step-1---create-an-azure-ad-app) to create an Azure AD app and a client secret.
+
+2. Give the app the [Reader Role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) for each data factory to access the REST APIs. Select `Access Control (IAM)` from each data factory's settings page and add the built-in `Reader` role to the app you created in Step 1.
```
Comment (mars-lan) [inline metaphor/azure_data_factory/readme.md:11]:
```suggestion
2. Grant the app the [Reader Role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) to each data factory. Select `Access Control (IAM)` from each data factory's settings page and add the built-in `Reader` role to the app you created in Step 1.
```

**Review by mars-lan — COMMENTED** [review]
Sorry one more edit.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**